ocs_ci/ocs/ui/page_objects/resource_list.py

```diff
@@ -0,0 +1,91 @@
+from selenium.webdriver.common.by import By
+
+from ocs_ci.ocs.exceptions import IncorrectUiOptionRequested
+from ocs_ci.ocs.ui.helpers_ui import format_locator
+from ocs_ci.ocs.ui.page_objects.searchbar import SearchBar
+from ocs_ci.ocs.ui.base_ui import logger
+
+
+class ResourceList(SearchBar):
+    """
+    Resource List module presented in OBC, OB, PVC, PV, BucketClass, BackingStore, StorageClass, VolumeSnapshotClasses,
+    BlockPools, etc.
+
+    This module is for selecting resource, navigation into, filtering, deletion, edition, etc.
+    """
+
+    def nav_to_resource(self, resource_name: str = None, resource_label: str = None):
+        """
+        Navigate to resource.
+        Important! Mandatory to pass either resource_name or resource_label
+        Args:
+            resource_name (str): Resource name - optional argument
+            resource_label (str): Resource label - optional argument
+        """
+        if not (resource_name or resource_label):
+            raise IncorrectUiOptionRequested(
+                "Either resource_name or resource_label should have value"
+            )
+
+        if resource_name:
+            logger.info(f"Navigate to resource by name '{resource_name}'")
+            self.select_search_by("name")
+            self.search(resource_name)
+            self.do_click(
+                format_locator(self.generic_locators["resource_link"], resource_name),
+                enable_screenshot=True,
+            )
+        elif resource_label:
+            logger.info(f"Navigate to resource by label '{resource_label}'")
+            raise NotImplementedError(
+                "TODO: select resource by label, first dropdown matching to entered label"
+            )
+
+    def select_resource_number(self, resource_number: int):
+        """
+        Select resource number
+        Args:
+            resource_number (int): Resource number (index)
+        """
+        logger.info(f"Select resource number {resource_number}")
+
+    def delete_resource(self, delete_via, resource):
+        """
+        Delete Object Bucket, Object Bucket Claim, PVC, PV, BucketClass, BackingStore, StorageClass, Namespace, etc.
+
+        Args:
+            delete_via (str): delete using 'three dots' icon, from the Resource List page
```

Comment:
It would be helpful to list here the supported values of _delete_via_

Reply:
added supported values